# HG changeset patch
# User Jacob Dawid <jacob.dawid@googlemail.com>
# Date 1335986108 -7200
#      Wed May 02 21:15:08 2012 +0200
# Branch gui
# Node ID 8e9823066dad4a6d894acd623403bf5fd3e80a12
# Parent  f6b4236da9f8beb9a513f6664fe520cadcd34459
Fixed bug with not correctly syncing settings.

MainWindow.cpp (processSettings): Settings dialog will be destroyed before the settingsChanged signal will be emitted.

diff --git a/gui/src/MainWindow.cpp b/gui/src/MainWindow.cpp
--- a/gui/src/MainWindow.cpp
+++ b/gui/src/MainWindow.cpp
@@ -191,18 +191,19 @@ void
 MainWindow::openOctaveForgePage ()
 {
   QDesktopServices::openUrl (QUrl ("http://octave.sourceforge.net/"));
 }
 
 void
 MainWindow::processSettingsDialogRequest ()
 {
-  SettingsDialog settingsDialog (this);
-  settingsDialog.exec ();
+  SettingsDialog *settingsDialog = new SettingsDialog (this);
+  settingsDialog->exec ();
+  delete settingsDialog;
   emit settingsChanged ();
   ResourceManager::instance ()->updateNetworkSettings ();
 }
 
 void
 MainWindow::showAboutOctave ()
 {
   QString message =
